geocoder/beam.cpp

```diff
@@ -0,0 +1,31 @@
+#include "geocoder/beam.hpp"
+
+#include "base/assert.hpp"
+
+#include <algorithm>
+
+namespace geocoder
+{
+Beam::Beam(size_t capacity) : m_capacity(capacity) { m_entries.reserve(m_capacity); }
+
+void Beam::Add(Key const & key, Value value)
+{
+  if (m_capacity == 0)
+    return;
+
+  Entry const e(key, value);
+  auto it = std::lower_bound(m_entries.begin(), m_entries.end(), e);
```

Comment:
BTW, entries are sorted, why not to use limited_priority_queue?

Reply:
Yes, it is possible. See further requests.